znai-core/src/main/java/com/twosigma/znai/reference/GlobalDocReferences.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019 TWO SIGMA OPEN SOURCE, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twosigma.znai.reference;
+
+import com.twosigma.znai.core.ResourcesResolver;
+
+import java.nio.file.Path;
+
+public class GlobalDocReferences {
+    private final Path globalReferencesPath;
+    private DocReferences docReferences;
+
+    private final ResourcesResolver resourcesResolver;
+
+    public GlobalDocReferences(ResourcesResolver resourcesResolver, Path globalReferencesPath) {
+        this.resourcesResolver = resourcesResolver;
+        this.globalReferencesPath = globalReferencesPath;
+        this.reload();
+    }
+
+    public void reload() {
+        docReferences = DocReferencesParser.parse(resourcesResolver.textContent(globalReferencesPath));
```

Comment:
Can you add `conteant` as a member variable into `DocReferencesParser` and inject `DocReferencesParser` instead of `resourcesResolver` and `globalReferencesPath`? 
You can call parser.parse() to get docReferences().

Reply:
not sure I follow.

Reply:
What I mean is that `GlobalDocReferences` doesn't need `resourcesResolver` and `globalReferencesPath`.
Instead you can inject `DocReferencesParser` into the constructor and keep it as a member variable.
`reload` will just call `docReferencesParser.parse()`.
To make the change, `DocReferencesParser` will have those member variables.

Also why don't `reload()` return `DocReferences`? Then you don't need `getDocReferences`.

Reply:
Actually you don't need this class if `DocReferencesParser` has the two member variables.

Reply:
I still need `get` in `com.twosigma.znai.website.WebSite#deployGlobalDocReferences`